src/main/java/org/datavyu/plugins/ffmpeg/MpvFxMediaPlayer.java

```diff
@@ -0,0 +1,117 @@
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URI;
+
+public final class MpvFxMediaPlayer extends MpvMediaPlayer {
+    private Stage stage;
+    private static final Logger LOGGER = LogManager.getFormatterLogger(MpvMediaPlayer.class);
+    private static final int JAVA_9 = 9;
+
+    public MpvFxMediaPlayer(URI mediaPath, Stage stage) {
+        super(mediaPath);
+        this.stage = stage;
+    }
+
+    @Override
+    public void init() {
+        addMediaPlayerStateListener(new PlayerStateListenerImpl());
+        initNative(); // starts the event queue, make sure to register all state/error listeners before
+
+        stage.show();
+
+        long[] newNativeMediaRef = new long[1];
+        int rc = mpvInitPlayer(newNativeMediaRef, mediaPath, getWindowId(stage));
+        if (0 != rc) {
+            throwMediaErrorException(rc, null);
+        }
+
+        nativeMediaRef = newNativeMediaRef[0];
+    }
+
+    /**
+     * Get the minor version of the java version triplet
+     *
+     * @return The minor version
+     */
+    private static int getMinorJavaVersion() {
+        String version = System.getProperty("java.version");
+        String[] triplet = version.split("\\.");
+        String minor = triplet[1];
+        return Integer.parseInt(minor);
+    }
+
+    /**
+     * Get the window id for a stage
+     * <p>
+     * This seems only to work on windows
+     * <p>
+     * Code adopted from https://github.com/java-native-access/jna/issues/706
+     *
+     * @param stage The stage
+     * @return The window id
+     * @throws RuntimeException if the window id can't be retrieved
+     */
+    private static long getWindowId(Stage stage) {
+        int minor = getMinorJavaVersion();
+        Class<?> stageClazz = stage.getClass();
+        try {
+            Method tkStageGetter = minor == JAVA_9 ? stageClazz.getSuperclass().getDeclaredMethod("getPeer")
+                    : stageClazz.getMethod("impl_getPeer");
+            tkStageGetter.setAccessible(true);
+            TKStage tkStage = (TKStage) tkStageGetter.invoke(stage);
+
+            Method getPlatformWindow = tkStage.getClass().getDeclaredMethod("getPlatformWindow");
+            getPlatformWindow.setAccessible(true);
+            Object platformWindow = getPlatformWindow.invoke(tkStage);
+
+            Method getNativeHandle = platformWindow.getClass().getMethod("getNativeHandle");
+            getNativeHandle.setAccessible(true);
+
+            Object nativeHandle = getNativeHandle.invoke(platformWindow);
+            return (long) nativeHandle;
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException ex) {
+            LOGGER.error("Unable to retrieve window id, due to Error: " + ex);
+            throw new RuntimeException("Unable to retrieve window id");
+        }
+    }
+
+    private class PlayerStateListenerImpl implements PlayerStateListener {
```

Comment:
Delete! Please review new commits in the PR (https://github.com/databrary/datavyu-ffmpegplugin/pull/105) we will have merge conflict since I pushed (and renamed) the PlayerStateListener to FfmpegMediaPlayer to avoid code redundancy.

Reply:
Can't delete this here because the implementation in 105 is based on the container not stage.